packages/vscode-extension/src/utilities/packageManager.ts

```diff
@@ -44,3 +56,68 @@ export function isPackageManagerAvailable(manager: PackageManagerName): boolean
   } catch {}
   return false;
 }
+
+async function isNpmInstalled(): Promise<boolean> {
+  const workspacePath = getAppRootFolder();
+  try {
+    const { stdout, stderr } = await command("npm ls --json", {
+      cwd: workspacePath,
+      silentErrorsOnExit: true,
+    });
+    const parsedJson = JSON.parse(stdout);
+    return parsedJson.problems ? false : true;
+  } catch (e) {
+    return false;
+  }
+}
+
+async function isYarnInstalled(): Promise<boolean> {
+  const workspacePath = getAppRootFolder();
+  try {
+    // because "yarn check" was removed from yarnv2 we use npm's method for checking dependencies
+    // npm is takeing into consideration yarn.lock since version 7, which means
+    // that if the users shell is using the older one this function may produce false in unexpected ways but even then
+    // we'll just run "yarn install" every time which is exactly what we would need to do without isYarnInstalled.
+    // https://docs.npmjs.com/cli/v7/commands/npm-install
+    const { stdout, stderr } = await command("npm ls --json", {
+      cwd: workspacePath,
+      silentErrorsOnExit: true,
+    });
+    const parsedJson = JSON.parse(stdout);
+
+    // because npm marks packages installed with yarn as "extraneous" we need to check if there are any other problems.
+    if (parsedJson.problems) {
+      parsedJson.problems.forEach((element: string) => {
+        if (!element.startsWith("extraneous")) {
+          return false;
+        }
+      });
+    }
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
+
+async function isPnpmInstalled(): Promise<boolean> {
+  // TODO: add pnpm support
+  return false;
+}
+
+async function isBunInstalled(): Promise<boolean> {
+  // TODO: add bun support
+  return false;
+}
+
+export async function isNodeModulesInstalled(manager: PackageManagerName): Promise<boolean> {
```

Comment:
This function names are confusing, I'd expect it checks for installation status of packages in node_modules but it uses `isXInstalled` inside which suggests only checking if some packager is available.

Reply:
isNpmModulesInstalled

Reply:
do you think something like that makes more senes?

Reply:
It's definitely better.